### ISSUE TITLE
Edge case comment_author_link, escape or not escape.

### ISSUE THREAD

**Issue by TukuToi** [WordPress/WordPress-Coding-Standards#2003]
See also https://github.com/WordPress/WordPress-Coding-Standards/issues/845#issuecomment-279834686

When using `comment_author_link`, WPCS complains that it should be escaped before echoing.

The only way to escape that would likely be a wp_kses, because everything else would result in either broken HTML or else.

It is confusing, because WordPress returns to us a full HTML set, 1/2 escaped
https://developer.wordpress.org/reference/functions/comment_author_link/

1. It builds that HTML with an escaped URL from https://developer.wordpress.org/reference/functions/get_comment_author_url/
2. But it does not escape or else sanitise the $author, as far I can see in https://developer.wordpress.org/reference/functions/get_comment_author/

Thus... do we have to escape `comment_author_link`, and if so, how?

**Comment by jrfnl** [comment]
@TukuToi General questions about how to write secure code should be posted to the WP forum. This repo is not a helpdesk for how to write plugins/themes.
Issues in this repos should be about problems with sniff correctness, the provided rulesets and/or feature requests.

If this issue is about whether or not a sniff is correct, it is missing essential information which is requested in the issues template, which you choose not to use, which makes this issue unactionable by the team here, no matter what.

> When using comment_author_link, WPCS complains that it should be escaped before echoing.

Not when you use it correctly.
```php
comment_author_link(); // Echos out a comment author link.
```

Note that the `comment_author_link()` function _echo_s out the result and you should not use `echo` in your code when using that function.
The sniff I suspect you are talking about `Security.EscapeOutput`, will only trigger when *your code* sends output.

Suggest: close.

**Comment by TukuToi** [comment]
Hi @jrfnl - and sorry!
Of course you are right.
I simply and dumbly assumed, the theme [here](https://de.wordpress.org/themes/blankslate/) "does it right" (see https://themes.svn.wordpress.org/blankslate/4.0.4/functions.php line `<li <?php comment_class(); ?> id="li-comment-<?php comment_ID(); ?>"><?php echo comment_author_link(); ?></li>`

My bad for not having researched this enough and getting confused by the other Git Hub issue I lined above.

Can be closed.
